src/core/components/core-tooltip.tsx

```diff
@@ -44,18 +46,22 @@ export function ChartTooltip({
   getTooltipContent: getTooltipContentOverrides,
   api,
   i18nStrings,
+  debounce = false,
 }: CoreChartProps.TooltipOptions & {
   i18nStrings?: BaseI18nStrings;
   getTooltipContent?: CoreChartProps.GetTooltipContent;
   api: ChartAPI;
 }) {
   const [expandedSeries, setExpandedSeries] = useState<ExpandedSeriesState>({});
   const tooltip = useSelector(api.tooltipStore, (s) => s);
-  if (!tooltip.visible || tooltip.group.length === 0) {
+  const debouncedTooltip = useDebouncedComponent(tooltip, debounce === true ? DEFAULT_DEBOUNCE : debounce || 0);
```

Comment:
The debouncedTooltip is not a component but a state. Should we rename the util to `useDebouncedValue` to avoid confusion? Likewise, inside the util we can have `const [debouncedValue, setDebouncedValue] = useState<T | null>(null);`